src/main/java/dev/unnm3d/rediseconomy/command/BackupRestoreCommand.java

```diff
@@ -37,20 +39,51 @@ public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command
             Path userPath = Path.of(plugin.getDataFolder().getAbsolutePath(), args[0]);
             switch (label) {
                 case "backup-economy" -> {
-                    try (FileWriter fw = new FileWriter(userPath.normalize().toFile())) {
-                        StringBuilder sb = new StringBuilder();
-                        currenciesManager.getCurrencies().forEach(currency ->
-                                currency.getAccounts().forEach((uuid, balance) ->
-                                        sb.append(currency.getCurrencyName())
-                                                .append(";")
-                                                .append(uuid.toString())
-                                                .append(";")
-                                                .append(currenciesManager.getUsernameFromUUIDCache(uuid))
-                                                .append(";")
-                                                .append(balance)
-                                                .append(System.getProperty("line.separator"))
-                                ));
-                        fw.write(sb.toString());// currency;uuid;name;balance
+                    final Map<UUID, String> names = new HashMap<>();
```

Comment:
Do you really need to check RAM requirements?
we are talking of megabytes not gigabytes.
i tested with like 40k entries.
for 400000 entries I calculated 57MB of RAM